[refactor] Add serializer_spec

donaldong opened the pull request:
This adds serializer_spec, fixes some rubocop lint errors, and delegates all the related array methods to execute queries.

donaldong committed:
[refactor] Add serializer_spec

Pull Request Branch: donaldong/refactor_add_serializer_spec
Pull Request Link: https://github.com/chanzuckerberg/redcord/pull/48

katyho commented on lib/redcord/attribute.rb:
```diff
@@ -58,6 +58,11 @@ def ttl(duration)
       class_variable_set(:@@ttl, duration)
     end
 
+    sig { returns(String) }
```
Why do you need to define this here? This is also defined in serializer

donaldong replied:
I moved this from serializer.rb to attribute.rb 🤔 I guess serializer.rb  is a better place for this method after all. Will put it back lol

katyho commented on lib/redcord/relation.rb:
```diff
@@ -67,24 +66,93 @@ def count
-  def to_a
-    execute_query
-  end
+  delegate(
```
Just wondering how you came up with this extensive list?

donaldong replied:
I went through https://ruby-doc.org/core-2.7.0/Array.html#method-i-7C and inspected existing methods on an array (with active support / array loaded). I selected most related methods to delegate.

katyho commented on spec/serializer_spec.rb:
```diff
@@ -0,0 +1,166 @@
+    expect(instance.updated_at.to_i).to eq(another_instance.updated_at.to_i)
+  end
+
+  it 'works with boolean types' do
```
What is the value of this test specifically? Is it just to check that a boolean value is serialized/deserialized correctly by the redis client?

donaldong replied:
yep!

katyho commented on spec/serializer_spec.rb:
```diff
@@ -0,0 +1,166 @@
+    expect(queried_instances.second.id).to eq(first.id).or eq(second.id)
+  end
+
+  it 'allows time types to be range index attributes' do
```
Is this a duplicate of the test above?

donaldong replied:
huh 🤔 idk how that happened. Good catch!

donaldong commented:
address comments

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/redcord/compare/fcd2617ddf302275f540eba663cbbc2e07cddb7c..bfa44d23a95cbc678e75135b8e0c79e5e62b88d6).
